java/memory/src/main/java/org/apache/arrow/memory/BufferAllocator.java

```diff
@@ -50,6 +50,27 @@
    */
   public ArrowBuf buffer(int size, BufferManager manager);
 
+  /**
+   * Allocate a new or reused buffer of the provided size. Note that the buffer may technically
+   * be larger or smaller than the  requested size, depending on the rounding option.
+   * However, the buffer's capacity will be set to the configured size.
+   *
+   * @param size The size in bytes.
+   * @param manager A buffer manager to manage reallocation.
+   * @param roundingOption The rounding option.
+   * @return a new ArrowBuf, or null if the request can't be satisfied
+   * @throws OutOfMemoryException if buffer cannot be allocated
+   */
+  public ArrowBuf buffer(int size, BufferManager manager, BaseAllocator.AllocationRoundingOption roundingOption);
```

Comment:
See question above, is changing rounding behavior dynamically something we specifically want to support, or can objects be created with the rounding option?

Reply:
@emkornfield Good question.
I think changing the rounding behavior dynamically is better, according to the analysis above.